modules/gmc-agent/src/main/java/org/gridgain/utils/QueryUtils.java

```diff
@@ -17,37 +17,61 @@
 package org.gridgain.utils;
 
 import org.apache.ignite.cache.query.QueryCursor;
+import org.apache.ignite.binary.BinaryObject;
+import org.apache.ignite.binary.BinaryObjectException;
+import org.apache.ignite.binary.BinaryType;
 import org.apache.ignite.cache.query.SqlFieldsQuery;
 import org.apache.ignite.internal.processors.cache.query.QueryCursorEx;
+import org.apache.ignite.internal.binary.BinaryObjectEx;
 import org.apache.ignite.internal.processors.query.GridQueryFieldMetadata;
+import org.apache.ignite.internal.util.IgniteUtils;
 import org.apache.ignite.internal.util.typedef.F;
 import org.apache.ignite.internal.util.typedef.internal.U;
-import org.apache.ignite.internal.visor.query.VisorQueryUtils;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.apache.ignite.internal.util.typedef.internal.SB;
 import org.gridgain.action.query.CursorHolder;
+import org.gridgain.action.query.QueryHolder;
 import org.gridgain.dto.action.query.QueryArgument;
 import org.gridgain.dto.action.query.QueryField;
 import org.gridgain.dto.action.query.QueryResult;
 
+import javax.cache.Cache;
+import java.math.BigDecimal;
+import java.net.URL;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.Arrays;
+import java.util.Date;
 import java.util.Iterator;
 import java.util.List;
 
 /**
  * SQL query utils.
  */
 public class QueryUtils {
+    /** Columns for SCAN queries. */
+    public static final List<QueryField> SCAN_COL_NAMES = Arrays.asList(
+        new QueryField().setFieldName("Key Class"), new QueryField().setFieldName("Key"),
+        new QueryField().setFieldName("Value Class"), new QueryField().setFieldName("Value")
+    );
+
     /**
      * @param curHolder Cursor id.
      * @param pageSize Page size.
      * @return Query result.
      */
-    public static QueryResult fetchResult(CursorHolder curHolder, int pageSize) {
+    public static QueryResult fetchResult(QueryHolder qryHolder, CursorHolder curHolder, int pageSize) {
```

Comment:
I think split this method on two will be better

Reply:
Done.